scopes/dependencies/dependency-resolver/dependency-resolver.docs.md

```diff
@@ -1,235 +1,251 @@
+
+For example, the following configuration will set `classnames` version `1.0.0` on all component using the `react-ui` namespace. This policy will override the workspace-level policy that uses version `2.0.0` of that same package.
+
+```json
+// All components using the namespace 'react-ui' will use version 1.0.0 of "classnames"
+// instead of version "2.0.0", set as the default for all components in the workspace
+{
+  "teambit.workspace/workspace": {
+    "name": "my-workspace"
+  },
+  "teambit.dependencies/dependency-resolver": {
+    "policy": {
       "dependencies": {
-        "underscore": "1.1.1"
-      },
-      "devDependencies": {
-        "types/underscore": "1.1.1"
+        "classnames": "2.0.0"
+      }
+    }
+  },
+  "teambit.workspace/variants": {
+    "teambit.dependencies/dependency-resolver": {
+      "policy": {
+        "{react-ui/*}": {
+          "classnames": "1.0.0"
+        }
       }
     }
   }
 }
 ```
 
-#### file dependencies definitions - TBD
-The dependency resolver will provide a hook called @FileDependencies (name is open - see open question below) to enable 3rd party extension analyze a file and return a list of dependencies from it.
-here is an example:
-```js
-// fileToAnalyze.ts
-import type {SomeType} from 'my-package';
-import default from 'my-component';
-
-console.log('do something');
-
-// my 3rd party extension
-import { Extension } from 'teambit.harmony/bit';
-import { DependencyResolver, FileDependencies } from 'teambit.dependencies/dependency-resolver';
-
-@Extension()
-export class MyExtension {
-  constructor() {}
-
-  getDepsForFile(filePath: string, fs: FS): FileDependenciesDefinition {
-    return [
-      {
-        dependencyPath: "my-package",
-        isType: true;
-      },
-      {
-        dependencyPath: "my-component"
+### "Forcibly" add dependencies to a component
+
+Dependency policies applied on a selected group of components will "forcibly" add the listed packages to any [selected] component that does not have them already listed as a dependency.
+This can be useful when a component depends on another module but has no `import`/`require` statement to be parsed by the Dependency Resolver (for example, in a Webpack configuration file).
+
+In the below example, `classnames@1.0.0` will be "forcibly" added as a dependency to any component using the `react-ui` namespace.
+
+```json
+  "teambit.workspace/variants": {
+    "teambit.dependencies/dependency-resolver": {
+      "policy": {
+        "{react-ui/*}": {
+          "classnames": "1.0.0"
+        }
+      }
+    }
+  }
+```
+
+### Setting dependency types
+
+#### Setting dev dependencies
+
+Dev dependencies are determined by the type of file that is dependent on them. If it is a development file (e.g, `doSomething.test.ts`) then the files consumed by it are also understood (by the Dependency Resolver) to be used for development and therefore, will be registered as `devDependencies`.
```

Comment:
mention that if something is both required in dev file and regular file it will be considered as runtime not a dev